ygen/codegen.go

```diff
@@ -549,16 +575,15 @@ func (dcg *DirectoryGenConfig) GetDirectoriesAndLeafTypes(yangFiles, includePath
 		return nil, nil, errs
 	}
 
-	dirsToProcess := map[string]*yang.Entry(mdef.directoryEntries)
-
-	enumSet, _, errs := findEnumSet(mdef.enumEntries, cg.TransformationOptions.CompressBehaviour.CompressEnabled(), false, cg.ParseOptions.SkipEnumDeduplication, cg.TransformationOptions.ShortenEnumLeafNames, cg.TransformationOptions.UseDefiningModuleForTypedefEnumNames, cg.GoOptions.AppendEnumSuffixForSimpleUnionEnums, false, cg.TransformationOptions.EnumOrgPrefixesToTrim)
+	enumSet, _, errs := findEnumSet(mdef.enumEntries, opts.TransformationOptions.CompressBehaviour.CompressEnabled(), !opts.TransformationOptions.EnumerationsUseUnderscores, opts.ParseOptions.SkipEnumDeduplication, opts.TransformationOptions.ShortenEnumLeafNames, opts.TransformationOptions.UseDefiningModuleForTypedefEnumNames, opts.AppendEnumSuffixForSimpleUnionEnums, opts.UseConsistentNamesForProtoUnionEnums, opts.TransformationOptions.EnumOrgPrefixesToTrim)
```

Comment:
(discussion) There are a lot of arguments to this function -- I wonder whether it's worth creating a `findEnumSetOpts` struct that is handed as an argument to try and make calling it a bit neater. The only reason to do this is probably readability, but it then comes with some of the concern of what happens if you don't consume a field (which is why we have it this way today). 

What are your thoughts on introducing such a struct?

Reply:
I think the readability is OK, maybe I got used to it though. I do think it's an advantage when the compiler tells you a parameter is unused.

Reply:
Sure, I'm good with that.